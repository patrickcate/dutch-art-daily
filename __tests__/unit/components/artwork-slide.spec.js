import { shallowMount } from '@vue/test-utils'
import ArtworkSlide from '@components/artwork-slide.vue'

describe('ArtworkSlide Component', () => {
  let artworkData

  beforeEach(() => {
    artworkData = {
      artwork: {
        id: '01-01',
        orientation: 'portrait',
      },
    }
  })

  it('is a Vue instance', () => {
    const wrapper = shallowMount(ArtworkSlide, {
      propsData: {
        ...artworkData,
      },
    })
    expect(wrapper.isVueInstance()).toBe(true)
  })

  it('renders correctly', () => {
    const wrapper = shallowMount(ArtworkSlide, {
      propsData: {
        ...artworkData,
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
