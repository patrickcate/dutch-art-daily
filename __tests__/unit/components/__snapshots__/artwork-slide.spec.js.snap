// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtworkSlide Component renders correctly 1`] = `
<li class="carousel__list-item">
  <div class="carousel__slide carousel__slide--portrait">
    <artwork-slide-image-stub id="01-01" class="carousel__slide-image artwork--portrait"></artwork-slide-image-stub>
  </div>
</li>
`;
